File: mobile/lib/models/meal_response.dart
class MealResponse {
  final int userId;
  final String mealType;
  final DateTime responseDate;
  final bool willAttend;

  MealResponse({
    required this.userId,
    required this.mealType,
    required this.responseDate,
    required this.willAttend,
  });

  Map<String, dynamic> toJson() => {
        'userId': userId,
        'mealType': mealType,
        'responseDate': responseDate.toIso8601String(),
        'willAttend': willAttend,
      };
}

File: backend/PgFoodSmart/Models/MealResponse.cs

// ---------------------------
// Models/MealResponse.cs
// ---------------------------

namespace PgFoodSmart.Models
{
    public class MealResponse
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string ? MealType { get; set; } // Dinner, Tiffin, Lunch
        public DateTime ResponseDate { get; set; }
        public bool WillAttend { get; set; }
    }
}